File: And/app/src/main/java/com/rc/demo/tablayoutviewpager/PagerAdapter.kt
package com.rc.demo.tablayoutviewpager

import android.content.Context
import  androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

class PagerAdapter(fm : FragmentManager) : FragmentPagerAdapter(fm) {

    var mNumOfTabs = 3

    fun PagerAdapter(fm : FragmentManager?, NumOfTabs : Int) {
        mNumOfTabs = NumOfTabs
    }

    override fun getPageTitle(position : Int) : CharSequence? {
        return "Tab-" + position
    }

    override fun getItem(position : Int) : Fragment {
        return when(position) {
            0 -> {
                tab1Page()
            }
            1 -> {
                tab2Page()
            }
            2 -> {
                tab3Page()
            }
            else -> tab3Page()
        }
    }

    override fun getCount() : Int {
        return mNumOfTabs
    }

}

File: iOS/TabBarV1/Swift/PagerAdapter.swift

import UIKit


final class PagerAdapter : RCPageViewControllerHelper {


    override init(_ fm: UIViewController){
        super.init(fm)
    }

    var mNumOfTabs: Int? = 3

    convenience init(_ fm: UIViewController?, _ NumOfTabs: Int){
        self.init(fm!)
        mNumOfTabs = NumOfTabs
    }

    override public func getPageTitle(_ position: Int) -> String?{
        return "Tab-" + position
    }

    override func getItem(_ position: Int) -> UIViewController{
        switch position{

        case 0:
            return tab1Page()

        case 1:
            return tab2Page()

        case 2:
            return tab3Page()

        default:
            print("Default case is required...")
            return tab3Page()
        
        }
    }

    override func getCount() -> Int?{
        return mNumOfTabs ?? 0
    }

}
